src/Type/Union/CommentAuthorUnionType.php

```diff
@@ -0,0 +1,86 @@
+<?php
+namespace WPGraphQL\Type\Union;
+
+use GraphQL\Type\Definition\UnionType;
+use WPGraphQL\Types;
+
+/**
+ * Class CommentAuthorUnionType
+ *
+ * In some situations, the type of term cannot be known until query time. The commentAuthorUnion allows for a 
+ * comment author to be queried and resolved to a number of types. Currently will return a user or commentAuthor.
+ *
+ * @package WPGraphQL\Type\Union
+ */
+class CommentAuthorUnionType extends UnionType {
+
+	/**
+	 * This holds an array of the possible types that can be resolved by this union
+	 * @var array
+	 */
+	private static $possible_types;
+
+	/**
+	 * CommentAuthorUnionType constructor.
+	 */
+	public function __construct() {
+
+		$config = [
+			'name' => 'commentAuthorUnion',
+			'types' => function() {
+				return self::getPossibleTypes();
+			},
+			'resolveType' => function( $value ) {
+			if ( ! empty( $value->user_id ) ) {
```

Comment:
indentation looks funky here

Reply:
Also, I screwed up on how this should look. . .the `$value` is actually what is passed from the field resolver that declares the union as the type. That is where we want to determine whether to return a WP_User or just pass through the `$comment` itself for the author Data to be plucked from.

So this can be changed to: 

```
resolveType' => function( $source ) {
    if ( $source instanceof \WP_User ) {
        $type = Types::user();
    } else if ( $source instanceof \WP_Comment ) {
        $type = Types::comment_author();
    } else {
        $type = null;
    }
    return $type;
}
```